packages/agreement-process/src/services/agreementService.ts

```diff
@@ -199,6 +206,17 @@ function retrieveAgreementDocument(
   return document;
 }
 
+export const retrieveDelegationById = async (
+  readModelService: ReadModelService,
+  delegationId: DelegationId
+): Promise<Delegation> => {
+  const delegation = await readModelService.getDelegationById(delegationId);
+  if (!delegation) {
+    throw delegationNotFound(delegationId);
+  }
+  return delegation;
+};
```

Comment:
Let's also add the kind here so that it's explicit what kind of delegation we are talking about, as did in #1210